src/providers/pdal/qgspdalprovider.cpp

```diff
@@ -71,7 +71,7 @@ QgsRectangle QgsPdalProvider::extent() const
 
 QgsPointCloudAttributeCollection QgsPdalProvider::attributes() const
 {
-  return mIndex->attributes();
+  return mIndex.get() ? mIndex->attributes() : QgsPointCloudAttributeCollection();
```

Comment:
Why not test directly `mIndex` for bool, instead of `mIndex.get()`?

Reply:
Just my preference